### ISSUE TITLE
Optimize Distributed LinearCrossEntropyLoss

### ISSUE THREAD

**Issue by pbontrager** [pytorch/torchtune#2693]
Taken from @nathan-az 's comment [here](https://github.com/pytorch/torchtune/pull/2667#discussion_r2076550101)

Materialisation of the full logit chunk on all TP workers can be avoided with either:

using PyTorch's [loss_parallel](https://docs.pytorch.org/docs/stable/distributed.tensor.parallel.html#torch.distributed.tensor.parallel.loss_parallel) - this would require slightly more change since the ctx manager needs to wrap the backward step, or
applying sequence-dim sharding, which should play nicely since CELoss is independent on that dimension
(2) might actually be relatively easy:

```
"output": ColwiseParallel(input_layouts=Shard(1), output_layouts=Replicate()),
```

Pretty sure dim 1 is the sequence dimension already, so if we output on that dimension as well I think we save on an all-to-all and the duplicated work. Option 2 could be done with DTensor.redistribute to avoid having to have a loss specific parallelize plan.
